examples/chat/src/dialer.js

```diff
@@ -13,27 +13,28 @@ async function run() {
   ])
 
   // Create a new libp2p node on localhost with a randomly chosen port
-  const peerDialer = new PeerInfo(idDialer)
-  peerDialer.multiaddrs.add('/ip4/0.0.0.0/tcp/0')
   const nodeDialer = new Node({
-    peerInfo: peerDialer
+    peerId: idDialer,
+    addresses: {
+      listen: ['/ip4/0.0.0.0/tcp/0']
+    }
   })
 
-  // Create a PeerInfo with the listening peer's address
-  const peerListener = new PeerInfo(idListener)
-  peerListener.multiaddrs.add('/ip4/127.0.0.1/tcp/10333')
+  // Add the listener peer multiaddr to the PeerStore
+  // This would be automatic if it was discovered through a discovery service
+  nodeDialer.peerStore.addressBook.set(idListener, [multiaddr('/ip4/127.0.0.1/tcp/10333')])
```

Comment:
Maybe it makes more sense to just add the peer id to the multiaddr and dial that instead of associating the two, then dialing the id? In practice, I don't see code being written the way it is now.